server/plugin.go

```diff
@@ -61,58 +66,84 @@ func (p *Plugin) OnActivate() error {
 		return errors.Wrap(err, "failed to ensure bot account")
 	}
 
+	p.conf = configurator.NewConfigurator(p.mm, p.BuildConfig, botUserID)
 	stored := api.StoredConfig{}
 	_ = p.mm.Configuration.LoadPluginConfiguration(&stored)
+	_ = p.conf.Reconfigure(&stored)
+	conf := p.conf.GetConfig()
 
-	awsClient := aws.NewAWSClient(stored.AWSAccessKeyID, stored.AWSSecretAccessKey, &mm.Log)
-
-	conf := configurator.NewConfigurator(mm, awsClient, p.BuildConfig, botUserID)
-	_ = conf.RefreshConfig(&stored)
-	store := store.New(mm, conf)
-	proxy := proxy.NewProxy(mm, awsClient, conf, store)
+	p.aws = aws.NewService(&p.mm.Log)
+	err = p.aws.Configure(conf)
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize AWS access")
+	}
 
-	mutex, err := cluster.NewMutex(p.API, mutexKey)
+	p.store = store.New(p.mm, p.conf)
+	// manifest store
+	mstore := p.store.Manifest()
+	mstore.InitBuiltin(
+		aws_hello.Manifest(),
+		builtin_hello.Manifest(),
+	)
+	err = mstore.Configure(conf)
 	if err != nil {
-		return errors.Wrapf(err, "failed creating cluster mutex")
+		return errors.Wrap(err, "failed to initialize the manifest store")
+	}
+	err = mstore.InitGlobal(p.aws.Client())
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize the global manifest list from marketplace")
+	}
+	// app store
+	appstore := p.store.App()
+	appstore.InitBuiltin(
+		builtin_hello.App(),
```

Comment:
why is that we do not include the `aws_hello` as a builtin app? (I guess the underlying q is: should all apps with builtin manifests  be automatically installed?)

Reply:
A `builtin` app is one that is compiled in, and implements `Upstream` directly. No AWS (nor an HTTP) app can be "builtin". In the manifests store, "built-in" is a misnomer since all that is builtin is a manifest, so in this case AWS Hello appears listed and is available to install. 2/5 we could get rid of "builtin manifests" and just add them to `manifest file` if needed, there is no present use-case except for the AWS Hello, and that one can just expose its manifest as `file:`.